src/frontend/src/lib/components/tokens/TokensSignedIn.svelte

```diff
@@ -21,10 +22,38 @@
 		({ id: tokenId }) =>
 			($balancesStore?.[tokenId]?.data ?? BigNumber.from(0n)).gt(0n) || displayZeroBalance
 	);
+
+	let sortingEnabled = true;
+	$: $pointerEventStore, (sortingEnabled = !$pointerEventStore);
+
+	const updateTokensToDisplay = ({
+		tokensToDisplay,
+		newTokensList
+	}: {
+		tokensToDisplay: Token[];
+		newTokensList: Token[];
+	}): Token[] => {
+		if (sortingEnabled) {
+			return newTokensList;
+		}
+
+		const tokenMap = new Map(
+			newTokensList.map((token) => [
+				`${token.id.description}-${token.network.id.description}`,
+				token
+			])
+		);
+		return tokensToDisplay.map(
+			(token) => tokenMap.get(`${token.id.description}-${token.network.id.description}`) ?? token
+		);
+	};
+
+	let tokensToDisplay: Token[] = [];
+	$: tokensToDisplay = updateTokensToDisplay({ tokensToDisplay, newTokensList: tokens });
```

Comment:
Does that work? Let's assume the user click on something and at the same time the tokens are updated. So the list is not updated ok but then, user ends their gesture. The tokens are still not updated. Is that the expected behavior?

Reply:
you are right, solved in last commit